### PR TITLE
Test Driver

### PR DESCRIPTION
**Pull request by dwightguth**
Here we update the test driver for the EVM tests so that we're capable of executing arbitrary IELE tests. This means passing function names and arguments to calls, as well as checking gas.

We also fix several minor bugs I found when testing ERC20.

### CODE REVIEW

**Comment by dwightguth** [comment]
@grosu please review briefly my ERC20 tests and the changes I made to the ERC20 contract itself.

@bmmoore @theo25  this is ready for review now. Can you please take a look?

**Comment by grosu** [comment]
LGTM overall.  I wish the test programs were actual readable IELE textual programs, but not a big deal now.  Maybe later.

**Comment by theo25** [comment]
These changes look good to me.

File: ethereum.md [hunk 1]
```diff
@@ -523,26 +532,33 @@ The `"transactions"` key loads the transactions.
...
            ( .Bag
           => <message>
-               <msgID>      !ID:Int                                 </msgID>
-               <txNonce>    #asUnsigned(#parseByteStack(TN))         </txNonce>
-               <txGasPrice> #asUnsigned(#parseByteStack(TP))         </txGasPrice>
-               <txGasLimit> #asUnsigned(#parseByteStack(TG))         </txGasLimit>
-               <sendto>     #asAccount(#parseByteStack(TT))          </sendto>
-               <value>      #asUnsigned(#parseByteStack(TV))         </value>
-               <v>          #asUnsigned(#parseByteStack(TW))         </v>
+               <msgID>      !ID:Int                              </msgID>
```
Comment (bmmoore) [inline ethereum.md]: Why is this showing up as a diff? I don't see any change in these lines.

Comment (dwightguth) [reply ethereum.md]: We added func and adjusted the whitespace.

**Pull request merged**